sqlmesh/core/macros.py

```diff
@@ -191,7 +192,13 @@ def __init__(
             elif v.is_import and getattr(self.env.get(k), c.SQLMESH_MACRO, None):
                 self.macros[normalize_macro_name(k)] = self.env[k]
             elif v.is_value:
-                self.locals[k] = self.env[k]
+                if k == c.SQLMESH_BLUEPRINT_VARS:
```

Comment:
Seems like a hack. Should we make a special `Executable` type instead?

Reply:
exactly what i was thinking

Reply:
I considered this, but the issue was that `c.SQLMESH_VARS` maps to a dict of primitives in the python environment. When we hydrate it from state, we just call `eval` on that dict and everything's fine because primitives can be evaluated. If it had an `Executable` value though wouldn't it crash?

Reply:
I've refactored this and moved blueprint vars under `c.SQLMESH_VARS` in the python env., discussing in Slack...